pkg/auth/handler/webapp/authflowv2/routes.go

```diff
@@ -322,7 +322,7 @@ func (n *AuthflowV2Navigator) navigateStepIdentify(s *webapp.AuthflowScreenWithF
 	}
 }
 func (n *AuthflowV2Navigator) navigateLoginStepAuthenticate(s *webapp.AuthflowScreenWithFlowResponse, r *http.Request, webSessionID string, result *webapp.Result) {
-	switch data := s.BranchStateTokenFlowResponse.Action.Data.(type) {
+	switch data := s.StateTokenFlowResponse.Action.Data.(type) {
```

Comment:
I see latest main uses `BranchStateTokenFlowResponse`, think should use `StateTokenFlowResponse` since `BranchStateTokenFlowResponse` is for showing available options only?

Reply:
It seems actually it is two case.
1. When you take the branch without actually sending an input: Use BranchStateTokenFlowResponse
2. When you take the branch which requires a actual input: Use `StateTokenFlowResponse`

Maybe need @louischan-oursky

Reply:
It seems using `StateTokenFlowResponse` here is correct.
When looking at `takeBranchResultSimple`, `StateTokenFlowResponse` is same as `BranchStateTokenFlowResponse` in case 1.